### PR TITLE
Support ZAPD bone enums for skeleton import

### PR DESCRIPTION
**Pull request by m000z0rz**
The Majora's Mask decomp uses limb enums on their assets, which changes the generated .c files for objects enough that skeletons won't import. This patch series adds a script to check how many files import skeletons from a decomp folder (currently only 3/176 or 1.7% successful) and adds handling for parsing these limb enums and using their when adding limbs. After these changes, 166/174 (95.4%) import skeletons without throwing exceptions.

I did most of my manual testing by:
1. Open blender
1. In Fast64, set the game to OOT
1. In the OOT settings under OOT File Settings, set the decomp path to a checkout of the Majora's Mask decomp (with `make setup` complete) so assets are extracted
1. Under the `OOT Skeleton Exporter` section, import a skeleton from the MM decomp.
    1. I typically tested with the Deku Buter using
        1. Skeleton: gDekuButlerSkel
        1. Object: object_dnj
        1. Actor Overlay: ovl_En_Dno

Alternatively, you can use the test script introduced in the first commit.
`cd` to `scripts/oot` and run

`python3 make_all_skeletons.py <path to your mm decomp> <output dir> ["1" to import animations too]`

For example, with the MM decomp checked out to ~/git/mm and with `make setup` complete there, I run

`python3 make_all_skeletons.py ~/git/mm test-out 1`

And when it completes, I can open the blend file generated at `scripts/oot/test-out/assets/objects/object_dnj/object_dnj.blend` to see the Deku Butler successfully imported with all its animations.

### CODE REVIEW

**Review by Yanis42 — COMMENTED** [review]
thanks for the PR! just small comments (also idk how to test this, like do I just import skeletons? and which one(s)?)

idk what the other contributors will think but imo we should start to introduce snake case for newer code from now (and change the existing one to make it consistent), just a detail though

also make sure there's no useless imports added in here

---
note that I don't know much about skeletons

**Comment by Yanis42** [comment]
oh also you should add a ``oot`` subfolder for the content of the ``scripts`` folder I think

File: fast64_internal/oot/oot_utility.py [hunk 1]
```diff
@@ -425,6 +468,16 @@ def ootGetObjectPath(isCustomExport, exportPath, folderName):
     return filepath
 
 
+def ootGetObjectHeaderPath(isCustomExport, exportPath, folderName):
```
[the remark this answers is not in the document]
Comment (m000z0rz) [reply fast64_internal/oot/oot_utility.py]: Ah yeah, I copied this from ootGetObjectPath (just above it). Added type hints to both

File: fast64_internal/oot/oot_utility.py [hunk 2]
```diff
@@ -183,6 +183,22 @@ def getOOTScale(actorScale: float) -> float:
     return bpy.context.scene.ootBlenderScale * actorScale
 
 
+def ootGetEnums(code: str) -> List["OOTEnum"]:
```
[the remark this answers is not in the document]
Comment (m000z0rz) [reply fast64_internal/oot/oot_utility.py:207]: Good question, no idea :) I moved the `OOTEnum` class to come just before `ootGetEnums` - does that looks good?

**Comment by m000z0rz** [comment]
Thanks for the review @Yanis42 ! I added testing instructions to the PR description and pushed updates that I think should resove the issues you found.

**Review by Yanis42 — APPROVED** [review]
seems to work properly

**Pull request merged**